pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -325,23 +342,6 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
 	}
 
-	// make sure to update the latest status even if there is an error later
-	defer func() {
```

Comment:
Do we have sufficient unit tests to cover this at a branch level?

Reply:
I assume all the unit tests are on UpdateStatefulSet, not the nested method?

Reply:
> Is this the only caller of that method?

Yes.

> I assume all the unit tests are on UpdateStatefulSet, not the nested method?

Yes, they are.

---

https://github.com/kubernetes/kubernetes/pull/109694 (the original PR) merged with some integration tests without unit tests. I'll add unit some tests for this.